### PR TITLE
simplify getting all models openai

### PR DESCRIPTION
**Pull request by dusvyat**
-remove ft_suffix
-simplify retreival of all models

## Description

Please include a summary of the change and the issue it solves. 

Fixes #issue_number

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/openai_handler/helpers.py [hunk 1]
```diff
@@ -111,37 +119,33 @@ def truncate_msgs_for_token_limit(messages, model_name, max_tokens, truncate='fi
...
         return num_tokens
     else:
-        raise NotImplementedError(f"""_count_tokens() is not presently implemented for model {model_name}.""")
+        raise NotImplementedError(
+            f"""_count_tokens() is not presently implemented for model {model_name}."""
+        )
 
 
-def get_available_models(api_key: str, all_models: List, finetune_suffix: Optional[str] = None) -> List[str]:
+def get_available_models(api_key: str) -> List[str]:
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/helpers.py:143]: change here

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 2]
```diff
@@ -31,12 +38,19 @@ def __init__(self, *args, **kwargs):
...
+            'default',
+            'conversational',
+            'conversational-full',
+            'image',
+            'embedding',
+        ]
         self.rate_limit = 60  # requests per minute
         self.max_batch_size = 20
         self.default_max_tokens = 100
-        self.all_models = ALL_MODELS
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py:39]: change here

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 3]
```diff
@@ -108,8 +140,7 @@ def create(self, target, args=None, **kwargs):
         args = args['using']
         args['target'] = target
         api_key = get_api_key('openai', args, self.engine_storage)
-        ft_suffix = self.engine_storage.json_get('ft-suffix')['ft-suffix']
-        available_models = get_available_models(api_key, self.all_models, ft_suffix)
+        available_models = get_available_models(api_key)
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py:143]: changte here

File: mindsdb/integrations/handlers/openai_handler/constants.py [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
+    + tuple(f'text-similarity-{model}-001' for model in COMPLETION_LEGACY_BASE_MODELS)
+    + tuple(f'text-search-{model}-query-001' for model in COMPLETION_LEGACY_BASE_MODELS)
+    + tuple(f'text-search-{model}-doc-001' for model in COMPLETION_LEGACY_BASE_MODELS)
+    + tuple(f'code-search-{model}-text-001' for model in COMPLETION_LEGACY_BASE_MODELS)
+    + tuple(f'code-search-{model}-code-001' for model in COMPLETION_LEGACY_BASE_MODELS)
+)
 
-IMAGE_MODELS = ('dall-e-3', )
-
-ALL_MODELS = list(set(CHAT_MODELS + COMPLETION_MODELS + COMPLETION_LEGACY_MODELS + EMBEDDING_MODELS + IMAGE_MODELS))  # noqa
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/constants.py:22]: change here

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 5]
```diff
@@ -45,33 +59,51 @@ def __init__(self, *args, **kwargs):
         try:
             self.engine_storage.json_get('ft-suffix')['ft-suffix']
         except (KeyError, TypeError):
-            self.engine_storage.json_set('ft-suffix', {'ft-suffix': binascii.b2a_hex(os.urandom(15)).decode()})
+            self.engine_storage.json_set(
+                'ft-suffix', {'ft-suffix': binascii.b2a_hex(os.urandom(15)).decode()}
+            )
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py]: remove ft-suffix

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 6]
```diff
@@ -569,8 +696,6 @@ def finetune(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = Non
             'training_file': train_file_id,
             'validation_file': val_file_id,
             'model': self._get_ft_model_type(prev_model_name),
-            'suffix': f'{ft_suffix}',
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py:572]: remove this, not needed

**Review by paxcema — APPROVED** [review]
LGTM, it's nice that we can drop this now that we don't offer a single public key :)

**Pull request merged**